.github/workflows/codeql.yml

```diff
@@ -0,0 +1,51 @@
+name: CodeQL
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    types:
+      - opened
+      - synchronize
+
+# When running CodeQL on pull requests, cancel in-progress jobs if another run is started for the
+# same pull request. This reduces our costs when you push multiple times in quick succession.
+concurrency:
+  # This concurrency group cancels in-progress jobs or runs on pull_request events only.
+  # github.head_ref is only defined on pull_request events. If github.head_ref is undefined, the
+  # concurrency group will fallback to the run ID, which is guaranteed to be both unique and
+  # defined for the run.
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+    timeout-minutes: 360
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Initialize CodeQL
+        uses: github/codeql-action/init@v2
+        with:
+          config-file: ./.github/codeql/codeql-config.yml
```

Comment:
need to copy this over too

Reply:
ah, that file can be empty in this repo, I'll remove the reference